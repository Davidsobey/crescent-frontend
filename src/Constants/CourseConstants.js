const CourseConstants = {
  CREATE_REQUEST: 'CREATE_COURSE_REQUEST',
  CREATE_SUCCESS: 'CREATE_COURSE_SUCCESS',
  CREATE_FAILURE: 'CREATE_COURSE_FAILURE',

  GETALL_REQUEST: 'GETALL_COURSE_REQUEST',
  GETALL_SUCCESS: 'GETALL_COURSE_SUCCESS',
  GETALL_FAILURE: 'GETALL_COURSE_FAILURE',

  GETUNSUBSCRIBED_REQUEST: 'GETUNSUBSCRIBED_COURSE_REQUEST',
  GETUNSUBSCRIBED_SUCCESS: 'GETUNSUBSCRIBED_COURSE_SUCCESS',
  GETUNSUBSCRIBED_FAILURE: 'GETUNSUBSCRIBED_COURSE_FAILURE',

  GETSUBSCRIBED_REQUEST: 'GETUNSUBSCRIBED_COURSE_REQUEST',
  GETSUBSCRIBED_SUCCESS: 'GETUNSUBSCRIBED_COURSE_SUCCESS',
  GETSUBSCRIBED_FAILURE: 'GETUNSUBSCRIBED_COURSE_FAILURE',

  DELETE_REQUEST: 'DELETE_COURSE_REQUEST',
  DELETE_SUCCESS: 'DELETE_COURSE_SUCCESS',
  DELETE_FAILURE: 'DELETE_COURSE_FAILURE',

  LOADCOURSE_REQUEST: 'LOADCOURSE_COURSE_REQUEST',
  LOADCOURSE_SUCCESS: 'LOADCOURSE_COURSE_SUCCESS',
  LOADCOURSE_FAILURE: 'LOADCOURSE_COURSE_FAILURE',

  EDIT_REQUEST: 'EDIT_COURSE_REQUEST',
  EDIT_SUCCESS: 'EDIT_COURSE_SUCCESS',
  EDIT_FAILURE: 'EDIT_COURSE_FAILURE',

  CLEAR_COURSES: 'CLEAR_COURSES',
};

export default CourseConstants;
