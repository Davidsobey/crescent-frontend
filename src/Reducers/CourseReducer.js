import { isNumber } from 'util';
import CourseConstants from '../Constants/CourseConstants';
import UserConstants from '../Constants/UserConstants';

function filterById(id, delId) {
  if (isNumber(id) && id !== 0 && id !== delId) {
    return true;
  }
  return false;
}

function CourseReducer(state = {}, action) {
  switch (action.type) {
    case CourseConstants.CREATE_REQUEST:
      return Object.assign({}, state, {
        newCourseId: undefined,
        creating: true,
      });
    case CourseConstants.CREATE_SUCCESS:
      return Object.assign({}, state, {
        newCourseId: action.newCourseId,
        creating: false,
      });
    case CourseConstants.CREATE_FAILURE:
      return Object.assign({}, state, {
        newCourseId: undefined,
        creating: false,
      });
    case CourseConstants.GETALL_REQUEST:
      return Object.assign({}, state, {
        courses: [],
        loading: true,
      });
    case CourseConstants.GETALL_SUCCESS:
      return Object.assign({}, state, {
        courses: action.courses,
        loading: false,
      });
    case CourseConstants.GETALL_FAILURE:
      return Object.assign({}, state, {
        courses: [],
        loading: false,
      });
    case CourseConstants.GETUNSUBSCRIBED_REQUEST:
      return Object.assign({}, state, {
        unsubscribed_courses: [],
        unsubscribed_courses_loading: true,
      });
    case CourseConstants.GETUNSUBSCRIBED_SUCCESS:
      return Object.assign({}, state, {
        unsubscribed_courses: action.courses,
        unsubscribed_courses_loading: false,
      });
    case CourseConstants.GETUNSUBSCRIBED_FAILURE:
      return Object.assign({}, state, {
        unsubscribed_courses: [],
        unsubscribed_courses_loading: false,
      });
    case CourseConstants.LOADCOURSE_REQUEST:
      return {
        course_loading: true,
      };
    case CourseConstants.LOADCOURSE_SUCCESS:
      return Object.assign({}, state, {
        course: action.course,
        course_loading: false,
      });
    case CourseConstants.LOADCOURSE_FAILURE:
      return {
        error: action.error,
        course_loading: false,
      };
    case CourseConstants.DELETE_REQUEST:
      return Object.assign({}, state, {
        courses: state.courses.filter(obj => filterById(obj.id, action.id)),
        loading: true,
      });
    case CourseConstants.DELETE_SUCCESS:
      return Object.assign({}, state, {
        loading: false,
      });
    case CourseConstants.DELETE_FAILURE:
      return Object.assign({}, state, {
        loading: false,
      });
    case CourseConstants.EDIT_REQUEST:
      return {
        course_editing: true,
      };
    case CourseConstants.EDIT_SUCCESS:
      return {
        course_editing: false,
      };
    case CourseConstants.EDIT_FAILURE:
      return {
        error: action.error,
        course_editing: false,
      };
    case CourseConstants.CLEAR_COURSES:
      return {
        newCourseId: state.newCourseId,
      };
    case CourseConstants.LOADCOURSES_REQUEST:
      return {
        loading: true,
      };
    case CourseConstants.LOADCOURSES_SUCCESS:
      return Object.assign({}, state, {
        courses: action.courses,
        loading: false,
      });
    case CourseConstants.LOADCOURSES_FAILURE:
      return {
        error: action.error,
        loading: false,
      };
    case UserConstants.LOGOUT:
      return {};
    default:
      return state;
  }
}

export default CourseReducer;
