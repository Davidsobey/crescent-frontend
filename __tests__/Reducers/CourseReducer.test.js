// Reducer to be tested
import CourseReducer from '../../src/Reducers/CourseReducer';
import CourseConstants from '../../src/Constants/CourseConstants';

describe('CourseReducer', () => {
  test('CREATE_SUCCESS', () => {
    const action = {
      type: CourseConstants.CREATE_SUCCESS,
      newCourseId: 1,
    };

    expect(CourseReducer(undefined, action)).toMatchSnapshot();
  });

  test('GETALL_SUCCESS', () => {
    const action = {
      type: CourseConstants.GETALL_SUCCESS,
      courses: [],
    };

    expect(CourseReducer(undefined, action)).toMatchSnapshot();
  });

  test('GETUNSUBSCRIBED_SUCCESS', () => {
    const action = {
      type: CourseConstants.GETUNSUBSCRIBED_SUCCESS,
      courses: [],
    };

    expect(CourseReducer(undefined, action)).toMatchSnapshot();
  });

  test('DELETE_SUCCESS', () => {
    const action = {
      type: CourseConstants.DELETE_SUCCESS,
    };

    expect(CourseReducer(undefined, action)).toMatchSnapshot();
  });

  test('LOADCOURSE_SUCCESS', () => {
    const action = {
      type: CourseConstants.LOADCOURSE_SUCCESS,
      course: {id:1, name: 'aaa'},
    };

    expect(CourseReducer(undefined, action)).toMatchSnapshot();
  });

  test('EDIT_SUCCESS', () => {
    const action = {
      type: CourseConstants.EDIT_SUCCESS,
    };

    expect(CourseReducer(undefined, action)).toMatchSnapshot();
  });

  test('LOADCOURSES_SUCCESS', () => {
    const action = {
      type: CourseConstants.LOADCOURSES_SUCCESS,
      courses: [],
    };

    expect(CourseReducer(undefined, action)).toMatchSnapshot();
  });
});