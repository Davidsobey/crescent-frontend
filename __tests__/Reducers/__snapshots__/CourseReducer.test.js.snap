// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseReducer CREATE_SUCCESS 1`] = `
Object {
  "creating": false,
  "newCourseId": 1,
}
`;

exports[`CourseReducer DELETE_SUCCESS 1`] = `
Object {
  "loading": false,
}
`;

exports[`CourseReducer EDIT_SUCCESS 1`] = `
Object {
  "course_editing": false,
}
`;

exports[`CourseReducer GETALL_SUCCESS 1`] = `
Object {
  "courses": Array [],
  "loading": false,
}
`;

exports[`CourseReducer GETUNSUBSCRIBED_SUCCESS 1`] = `
Object {
  "unsubscribed_courses": Array [],
  "unsubscribed_courses_loading": false,
}
`;

exports[`CourseReducer LOADCOURSE_SUCCESS 1`] = `
Object {
  "course": Object {
    "id": 1,
    "name": "aaa",
  },
  "course_loading": false,
}
`;

exports[`CourseReducer LOADCOURSES_SUCCESS 1`] = `
Object {
  "subscribedCourses": Array [],
  "subscribedCourses_loading": false,
}
`;
